py/desispec/qa/qa_plots.py

```diff
@@ -156,30 +159,99 @@ def frame_skyres(outfil, frame, skymodel, qaframe):
 
     # Histogram of all residuals
     ax1 = plt.subplot(gs[1,0])
-    binsz = 0.1
     xmin,xmax = -5., 5.
-    i0, i1 = int( np.min(devs) / binsz) - 1, int( np.max(devs) / binsz) + 1
-    rng = tuple( binsz*np.array([i0,i1]) )
-    nbin = i1-i0
+
+    ##binsz = 0.1
+    ##i0, i1 = int( np.min(devs) / binsz) - 1, int( np.max(devs) / binsz) + 1
+    ##rng = tuple( binsz*np.array([i0,i1]) )
+    ##nbin = i1-i0
     # Histogram
-    hist, edges = np.histogram(devs, range=rng, bins=nbin)
+    ##hist, edges = np.histogram(devs, range=rng, bins=nbin)
+
+    binsz = qaframe.qa_data['SKYSUB']["PARAMS"]["BIN_SZ"]
+    hist = np.asarray(qaframe.qa_data['SKYSUB']["METRICS"]["DEVS_1D"])
+    edges = np.asarray(qaframe.qa_data['SKYSUB']["METRICS"]["DEVS_EDGES"])
+
     xhist = (edges[1:] + edges[:-1])/2.
     #ax.hist(xhist, color='black', bins=edges, weights=hist)#, histtype='step')
     ax1.hist(xhist, color='blue', bins=edges, weights=hist)#, histtype='step')
     # PDF for Gaussian
-    area = len(devs) * binsz
+    ##area = len(devs) * binsz
+
+    area = binsz * np.sum(hist)
+
     xppf = np.linspace(scipy.stats.norm.ppf(0.0001), scipy.stats.norm.ppf(0.9999), 100)
     ax1.plot(xppf, area*scipy.stats.norm.pdf(xppf), 'r-', alpha=1.0)
 
     ax1.set_xlabel(r'Res/$\sigma$')
     ax1.set_ylabel('N')
     ax1.set_xlim(xmin,xmax)
 
-
     # Meta text
+    #- limit the dictionary to residuals only for meta
+    qaresid=copy.deepcopy(qaframe)
```

Comment:
We started to replace with a shallow copy but ran into the problem we
 had originally where this keeps the references to subsidiary  
 information in the original compound object.  This is what we are
    trying to avoid by doing the copy, so we have left it.  If a copy() is 
    added to the Frame class, we can certainly use that later, but for now
    we've reverted to the deepcopy usage.  There are other places in the
    pipeline where deepcopy is used.